packages/portal-contracts/src/Portal.sol

```diff
@@ -414,6 +421,42 @@ contract Portal is Owned {
         _transferToSender(tokToSend);
     }
 
+    // Update an existing bid or ask order. amountSats is negative if the order is an ask.
+    function updateOrder(uint256 orderID, int128 amountSats) public payable {
+        Order storage o = orderbook[orderID];
+
+        require(o.amountSats != 0, "Order not found");
+        require(o.maker == msg.sender, "Order not yours");
+
+        if (o.amountSats < 0) {
+          // This is an ask. Receive the ether to be sold and update the order.
+          require(amountSats < 0 && uint256(int256(-(o.amountSats + amountSats))) <= MAX_SATS, "Must be adding a valid amount of liquidity");
```

Comment:
I'd calculate the new amount `o.amountSats + amountSats` first.

Gotta check that it's still negative, otherwise this function lets a user convert a bid into an ask.

Reply:
If 0.amountSats and amountSats are the same sign, is that something we have to worry about?

Maybe in the overflow case, but since we're on solidity 0.8+ that's checked automatically AFAIK

Reply:
I see--so the goal is to only let people add liquidity?

Then, for an existing order, you can increase the size via `updateOrder`, but if you want to decrease the size or change the price, you have to cancel + place a new order.

LMK if I understood it right